tla/ccfraft.tla

```diff
@@ -104,8 +104,18 @@ ASSUME Servers \subseteq AllServers
 VARIABLE reconfigurationCount
 \* Each server keeps track of the pending configurations
 VARIABLE configurations
-
-reconfigurationVars == <<reconfigurationCount, configurations>>
+\* The set of servers that have been removed from configurations.  The implementation
+\* assumes that a server refrains from rejoining a configuration if it has been removed
+\* from an earlier configuration (relying on the TEE and absent Byzantine fault). Here,
+\* we model removedFromConfiguration as a global variable. In the implementation, this
+\* state has to be maintained by each node separately.
+\* Note that we cannot determine the removed servers from configurations because a prefix
+\* of configurations is removed from the configuration on a change of configuration.
+\* TODO: How does the implementation keep track of removed servers?  Can we remove and
```

Comment:
The implementation does not explicitly track removed servers. If a node fails, then it rejoins with a new identity (and thus a new node ID). Node ID are unique (assuming a collision resistant hash function etc).

Reply:
The way I think about it, node rejoining is a specification issue that exists because the current specification only covers a particular aspect of CCF and could do a better job at capturing the assumptions it is making about the rest of CCF that is not specified in TLA+.